pkg/server/datastore/datastore.go

```
@@ -2,319 +2,624 @@ package datastore
+	if err != nil {
+		return nil, err
+	}
+
+	params := UpdateTrustDomainParams{
+		ID:               pgID,
+		OnboardingBundle: req.OnboardingBundle,
+	}
 
-type MemStore struct {
-	members      map[string]*Member // trust_domain (e.g. 'example.org') -> member
-	relationship []*Relationship
-	tokens       map[string]*AccessToken // token uuid string -> access token
+	if req.Description != "" {
+		params.Description = sql.NullString{
+			String: req.Description,
+			Valid:  true,
+		}
+	}
 
-	mu sync.RWMutex
+	if !req.HarvesterSpiffeID.IsZero() {
+		params.HarvesterSpiffeID = sql.NullString{
+			String: req.HarvesterSpiffeID.String(),
+			Valid:  true,
+		}
+	}
+
+	trustDomain, err := d.querier.UpdateTrustDomain(ctx, params)
+	if err != nil {
+		return nil, fmt.Errorf("failed updating trust domain: %w", err)
+	}
+	return &trustDomain, nil
 }
 
-func NewMemStore() DataStore {
-	return &MemStore{
-		members: make(map[string]*Member),
-		tokens:  make(map[string]*AccessToken),
-		mu:      sync.RWMutex{},
+func (d *SQLDatastore) DeleteTrustDomain(ctx context.Context, trustDomainID uuid.UUID) error {
+	pgID, err := uuidToPgType(trustDomainID)
+	if err != nil {
+		return err
+	}
+
+	if err = d.querier.DeleteTrustDomain(ctx, pgID); err != nil {
+		return fmt.Errorf("failed deleting trust domain with ID=%q: %w", trustDomainID, err)
 	}
+
+	return nil
 }
 
-func (s *MemStore) CreateMember(_ context.Context, member *common.Member) (*Member, error) {
-	s.mu.Lock()
-	defer s.mu.Unlock()
+func (d *SQLDatastore) ListTrustDomains(ctx context.Context) ([]*entity.TrustDomain, error) {
+	trustDomains, err := d.querier.ListTrustDomains(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed getting trust domain list: %w", err)
+	}
+
+	result := make([]*entity.TrustDomain, len(trustDomains))
+	for i, m := range trustDomains {
+		r, err := m.ToEntity()
+		if err != nil {
+			return nil, fmt.Errorf("failed converting model trust domain to entity: %v", err)
```

Total ignorance from my part. Shouldn't these errors trigger a log entry via the standardized library we are using (logrus)?

---

"Log and return the error" is an anti-pattern. It's better to return the error to the client and let it decide what to do with it.